src/csm_testing/tests/sat_functional/firmware/test_firmware.py

```diff
@@ -0,0 +1,188 @@
+        proc = subprocess.run(shlex.split(command), stdout=subprocess.PIPE, stderr=subprocess.PIPE, check=True)
+        output = proc.stdout.decode().strip()
+        info_msg = proc.stderr.decode().strip()
+
+        # Extract the snapshot name from the info message
+        pattern = (
+            r'INFO: Snapshot (\S+) created successfully\.\n'
+            r'INFO: Use `sat firmware --delete-snapshot` to delete the snapshot if it is no longer needed'
+        )
+        match = re.search(pattern, info_msg)
+        self.assertIsNotNone(match, "Snapshot creation message not found in info_msg.")
+        snapshot_name = match.group(1)
+
+        # Check that the output contains the expected table header
+        for col in SAT_FIRMWARE_HEADER:
+            self.assertIn(col, output)
+
+        # Now delete the snapshot to keep the system clean
+        delete_command = f"sat firmware --delete-snapshot {snapshot_name}"
+        proc = subprocess.run(shlex.split(delete_command), stdout=subprocess.PIPE, stderr=subprocess.PIPE, check=True)
+        delete_output = proc.stderr.decode().strip()
+        expected_msg = f"INFO: Snapshot {snapshot_name} deleted successfully."
+        self.assertIn(expected_msg, delete_output)
+
+    def test_firmware_describe_existing_snapshot(self):
+        """Test that `sat firmware --snapshots SNAPSHOT` describes an existing firmware snapshot."""
+        # Get the list of available snapshots
+        list_command = "sat firmware --snapshots"
+        proc = subprocess.run(shlex.split(list_command), stdout=subprocess.PIPE, stderr=subprocess.PIPE, check=True)
+        snapshots = [line.strip() for line in proc.stdout.decode().splitlines() if line.strip() and not line.startswith("#")]
+        self.assertGreater(len(snapshots), 0, "No firmware snapshots available for testing.")
+
+        # Query the first snapshot
+        snapshot = snapshots[0]
+        query_command = f"sat firmware --snapshots {snapshot}"
+        proc = subprocess.run(shlex.split(query_command), stdout=subprocess.PIPE, stderr=subprocess.PIPE, check=True)
+        output = proc.stdout.decode().strip()
+
+        # Check that the output contains the snapshot name and the expected table header
+        self.assertIn(snapshot, output)
+        for col in SAT_FIRMWARE_HEADER:
+            self.assertIn(col, output)
+
+    def test_firmware_query_multiple_snapshots(self):
+        """Test that `sat firmware --snapshots SNAPSHOT_1 SNAPSHOT_2` prints the details for two snapshots."""
+        # Get the list of available snapshots
+        list_command = "sat firmware --snapshots"
+        proc = subprocess.run(shlex.split(list_command), stdout=subprocess.PIPE, stderr=subprocess.PIPE, check=True)
+        snapshots = [line.strip() for line in proc.stdout.decode().splitlines() if line.strip() and not line.startswith("#")]
+        self.assertGreaterEqual(len(snapshots), 2, "Less than two firmware snapshots available for testing.")
```

Comment:
I would suggest making sure the test is self-contained by creating the needed snapshots in this test method or in the `setUpClass` method. You can make them small snapshots by using the `--xname` option to ensure good performance.